Kernel/Arch/aarch64/Processor.cpp

```diff
@@ -109,8 +109,12 @@ void Processor::initialize(u32)
         asm volatile("wfi");
 }
 
-void Processor::flush_tlb_local(VirtualAddress, size_t)
+void Processor::flush_tlb_local(VirtualAddress vaddr, size_t page_count)
 {
+    // NOTE: We must flush the data cache here, such that the updated page tables actually
+    //       end up in main memory, and thus actually get used by the MMU.
+    Aarch64::Asm::flush_data_cache(vaddr.get(), 4096 * page_count);
+
     // FIXME: Figure out how to flush a single page
```

Comment:
The code you added fixes the FIXME here, yes? Or no.

Reply:
Nope, that FIXME is about the TLB flushing below, which still flushes the complete TLB caches. I know it's kinda confusing, but there is a difference between the data cache and the TLB cache. The TLB cache just caches the result of virtual->physical translations, such that successive translations can use the cached result.